### PR TITLE
Add label and asterisk to textarea and datepicker

### PR DESCRIPTION
**Pull request by sdevalapurkar-bcgov**
## Overview

Add label and asterisk to textarea and datepicker

### CODE REVIEW

File: src/components/date-pick/DatePick.mdx [hunk 1]
```diff
@@ -9,17 +9,21 @@ Date picker component for allowing users to select a date.
...
+| label           | string   | false    | label text to display above the date picker                  | ""      |
+| isRequired      | boolean  | false    | determines whether date picker selection is mandatory or not | false   |
 
 <br />
 
 ## Default Date Picker
 
 Used for selecting a date.
 
+<Story name="Default" id="date-pick--default"></Story>
```
Comment (akroon3r) [inline src/components/date-pick/DatePick.mdx:25]: double dash in the id

File: src/components/date-pick/DatePick.mdx [hunk 2]
```diff
@@ -29,4 +33,17 @@ const [selectedDate, setSelectedDate] = useState(new Date('August 19, 1975 23:15
 />
 ```
 
-<Story name="Default" id="date-pick--default"></Story>
+## With Label (Mandatory) Date Picker
+
+Used when selecting a date is mandatory.
+
+<Story name="With Label (Mandatory)" id="date-pick--with-label"></Story>
```
Comment (akroon3r) [inline src/components/date-pick/DatePick.mdx:40]: double dash in id here too

File: src/components/date-pick/__snapshots__/DatePick.test.js.snap [hunk 3]
```diff
@@ -1,48 +1,116 @@
...
+    >
+      Select date
+    </label>
+    <span
+      className="mandatory"
+      id="asterisk"
+    >
+      *
+    </span>
+
```
Comment (akroon3r) [inline src/components/date-pick/__snapshots__/DatePick.test.js.snap:69]: whitespace

File: src/components/textarea/__snapshots__/Textarea.test.js.snap [hunk 4]
```diff
@@ -10,6 +10,8 @@ Array [
     >
       some label
     </label>
+
```
Comment (akroon3r) [inline src/components/textarea/__snapshots__/Textarea.test.js.snap:13]: whitespace x 2

**Comment by sdevalapurkar-bcgov** [comment]
all those things are expected :)

**Comment by akroon3r** [comment]
🧹 consider this...linted

**Pull request merged**